File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<vector<int>> bucket(nums.size()+1);
         unordered_map<int,int> hm;
         vector<int> res;
      
        for(int n:nums){
                hm[n]++;
        }

        for(auto& [num,freq]:hm){
            bucket[freq].push_back(num);
        }
        
        for(int j=nums.size();j>=0 && k>0;--j){
           for(int n: bucket[j]){
              res.push_back(n);
              k--;
            if (k<0){
                break;
            }
            }

        }
        return res;
    }
};

File: Data Structures & Algorithms/arranging-coins/submission-1.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        li=[]
        i=0
        res=0
        while n!=0:
            k=i+1
            
            if k<=n :
                li.append(k)
            else:
                break
            n-=(i+1)
            i+=1

        for i in li:
            res+=1
            
        return res
                

